Classes/Provider/CloudflareProxyProvider.php

```diff
@@ -14,39 +14,28 @@
 
 use B13\CloudflareCDN\CloudflareClient;
 use B13\Proxycachemanager\Provider\ProxyProviderInterface;
+use Psr\Http\Message\RequestInterface;
 use TYPO3\CMS\Core\Utility\GeneralUtility;
 
 /**
  * Adapter to EXT:proxycachemanager
  */
 class CloudflareProxyProvider implements ProxyProviderInterface
 {
-    /**
-     * @var CloudflareClient
-     */
-    protected $client;
+
+    protected CloudflareClient $client;
 
     public function __construct()
     {
         $this->client = GeneralUtility::makeInstance(CloudflareClient::class);
```

Comment:
DI?

Reply:
hm, i think this is not a good idea, have a look on how thi ProxyProvider is build https://github.com/b13/proxycachemanager/blob/master/Classes/ProxyConfiguration.php#L40 (bei GeneralUtility), if we want to use DI here, we have to register the provider as a public service, the provider is already a singleton service when retrived from the ProxyConfiguration, see implementation of the getProxyProvider() method